### PR TITLE
Add troubleshooting page

### PR DESCRIPTION
**Pull request by johnnycase**
site/content/docs: Add troubleshooting page
Created a troubleshooting page and added a troubleshooting section to the main Docs page.

Updates #3816 
Signed-off-by: johnnycase <johneverettcase@gmail.com>

### CODE REVIEW

File: site/content/docs/main/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+## Troubleshooting
+
+If you encounter issues, follow the guides below for help. For topics not covered here, you can [file an issue][0], or talk to us on the [#contour channel][1] on the Kubernetes Slack server.
```
Comment (stevesloka) [inline site/content/docs/main/troubleshooting.md:3]:
```suggestion
If you encounter issues, follow the guides below for help. For topics not covered here, you can [file an issue][0], or talk to us on the [#contour channel][1] on Kubernetes Slack.
```

**Review by stevesloka — CHANGES_REQUESTED** [review]
Just one small nit, overall looks like a good start to build out the different sections. 

We might also want to include the FAQ (https://projectcontour.io/resources/faq/) page to link to or merge together someway.

File: site/content/docs/main/troubleshooting.md (hunk 1 above)
Comment (johnnycase) [reply site/content/docs/main/troubleshooting.md:3]: Looks good to me, thanks!

**Review by youngnick — APPROVED** [review]
This looks great, thanks.

**Comment by youngnick** [comment]
Ah, it looks like your most recent commit needs the signoff @johnnycase. Personally, I use `git commit --amend -as` to fix this sort of thing up, that will update the last commit to have the signed-off-by line. You'll need to `git push --force` afterwards too. Or, if you did that with the online edit, unfortunately you need to fix this by pulling, editing locally, and repushing. Sorry, this is a Github limitation.

It's actually better to do stuff locally rather than use the UI for this reason, sadly.

Once the CI passes and the DCO is sorted, I'll merge this for you.

**Comment by johnnycase** [comment]
i see, i'll make those edits and resubmit, thanks!

**Comment by youngnick** [comment]
Dang, it looks like the second commit here is still not signed off. You could squash, or if you're not up for that, I can just squash as we merge.

**Comment by johnnycase** [comment]
Thanks for merging! Sorry about the commit not being signed off. The command I used was git commit -s -m "Message" should I add anything there?

**Pull request merged**